Tests/FoundationEssentialsTests/GregorianCalendarRecurrenceRuleTests.swift

```diff
@@ -739,4 +739,16 @@ final class GregorianCalendarRecurrenceRuleTests: XCTestCase {
         XCTAssertEqual(dates.next(), Date(timeIntervalSince1970: 1767484800.0)) // 2026-01-04T00:00:00-0000
         XCTAssertEqual(dates.next(), Date(timeIntervalSince1970: 1799020800.0)) // 2027-01-04T00:00:00-0000
     }
+
+    func testRangeExcludesUpperBounds() {
+         let start = Date(timeIntervalSince1970: 1695304800.0) // 2023-09-21T14:00:00-0000
+         let end   = Date(timeIntervalSince1970: 1729519200.0) // 2024-10-21T14:00:00-0000
+
+         let rule = Calendar.RecurrenceRule(calendar: gregorian, frequency: .monthly)
+
+         let eventStart = Date(timeIntervalSince1970: 1285077600.0) // 2010-09-21T14:00:00-0000
+         let results = Array(rule.recurrences(of: eventStart, in: start..<end))
+
+        XCTAssertNotEqual(results.last!, end)
```

Comment:
Instead of doing XCTAssertNotEqual, can you XCTAssertEqual what the last result should be?

Reply:
I can, but that changes the intent of the test, which is to ensure that `end` is not part of the recurrence. What the last value is doesn't actually matter for the purpose of this test.

Reply:
How about change the assertion to `XCTAssertLessThan`?

Reply:
I understand what you want to test, but this line, `XCTAssertNotEqual(results.last!, end)`, alone can mean anything if I hadn't looked at the code change. For example, this assert would pass if `results` is just an arbitrary value. Same goes with `XCTAssertLessThan`.

Also, instead of force unwrapping `results.last!`, it's useful to write it as `XCTUnwrap` instead of just force unwrapping it so we get a nice test report if it fails

Reply:
I was unaware of XCTUnwrap! That's neat! I'll make the changes later today!

Reply:
I decided to go with a pattern established elsewhere and just use a smaller list of recurrences and validate the entire array. Let me know if that doesn't work for some reason.